client/main.lua

```diff
@@ -1,206 +1,218 @@
-                StopScreenEffect("RaceTurbo")
-                TriggerServerEvent('qbx_nitro:server:UnloadNitrous', Plate)
-                nitrousActivated = false
-            end
-            if IsControlJustReleased(0, 36) and cache.seat == -1 then
-                SetVehicleBoostActive(cache.vehicle, 0)
-                SetVehicleEnginePowerMultiplier(cache.vehicle, 1.0)
-                SetVehicleEngineTorqueMultiplier(cache.vehicle, 1.0)
-                for index, _ in pairs(Fxs) do
-                    StopParticleFxLooped(Fxs[index], 1)
-                    TriggerServerEvent('qbx_nitro:server:StopSync', Plate)
-                    Fxs[index] = nil
+                StartScreenEffect('RaceTurbo', 0, false)
+                vehicleState:set('nitro', vehicleState.nitro - 0.25, true)
+                vehicleState:set('nitroPurge', (vehicleState.nitroPurge or 0) + 1, true)
+                if vehicleState.nitroPurge >= 100 then
+                    exports.qbx_core:Notify(locale('notify.needs_purge'), 'error')
+                    stopBoosting()
                 end
-                StopScreenEffect("RaceTurbo")
-                nitrousActivated = false
+            else
+                stopBoosting()
+                vehicleState:set('nitro', 0, true)
             end
-            Wait(0)
+            Wait(100)
         end
     end)
 end
 
-local function nitrousLoop()
-    local sleep = 0
+local function stopPurging()
+    local vehicleState = Entity(cache.vehicle).state
+    vehicleState:set('purgeNitro', false, true)
+    PurgeLoop = false
+end
+
+local function nitrousPurgeLoop()
+    local vehicleState = Entity(cache.vehicle).state
+    PurgeLoop = true
     CreateThread(function()
-        while cache.vehicle do
-            if (Entity(cache.vehicle)?.state?.nitro or 0) > 0 then
-                sleep = 0
-                if IsControlJustPressed(0, 36) and cache.seat == -1 and not nitroDelay then
-                    TriggerServerEvent('qbx_nitro:server:SyncFlames', VehToNet(cache.vehicle))
-                    nitrousActivated = true
-                    loadEffects()
-                    nitrousUseLoop()
-                end
+        while PurgeLoop and cache.vehicle do
+            if vehicleState.nitroPurge - 1 >= 0 then
+                vehicleState:set('nitroPurge', vehicleState.nitroPurge - 1, true)
             else
-                sleep = 1000
+                vehicleState:set('nitroPurge', 0, true)
+                stopPurging()
             end
-            Wait(sleep)
+            Wait(100)
         end
     end)
 end
 
-lib.onCache('vehicle', function(vehicle)
-    if vehicle and (not config.turboRequired or IsToggleModOn(vehicle, 18)) then
-        SetTimeout(750, function()
-            nitrousLoop()
-        end)
-    end
+qbx.entityStateHandler('nitroFlames', function(veh, netId, value)
+    if not veh or not DoesEntityExist(veh) then return end
+
+    lib.requestNamedPtfxAsset("veh_xs_vehicle_mods")

```

Comment:
I see that you load PtfxAsset but I cant find where you unload it?

Make sure that happens otherwise we might get issues with waitFor and other resources requesting ptfx

Reply:
I don't think it's even needed actually lol